pkg/controller/cloud/nodecontroller.go

```diff
@@ -91,10 +109,79 @@ func (cnc *CloudNodeController) Run() {
 	go func() {
 		defer utilruntime.HandleCrash()
 
+		instances, ok := cnc.cloud.Instances()
+		if !ok {
+			utilruntime.HandleError(fmt.Errorf("failed to get instances from cloud provider"))
+			return
+		}
+
+		// Start a loop to periodically update the node addresses obtained from the cloud
 		go wait.Until(func() {
```

Comment:
don't hate me, but I think it would be nice to break this rather large function out into a top-level func, and pass-in any args needed - it's too long for an inline definition.